### PR TITLE
Fixed #25583 -- Allowed calling `transform` with `CoordTransform` even if SRID is invalid.

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://code.djangoproject.com/ticket/25583

### CODE REVIEW

File: docs/ref/contrib/gis/geos.txt [hunk 1]
```diff
@@ -553,6 +553,8 @@ is returned instead.
 
     Requires GDAL. Raises :class:`~django.contrib.gis.geos.GEOSException` if
     GDAL is not available or if the SRID is ``None`` or less than 0.
+    Imposes no constraints on SRID if called with a
+    :class:`~django.contrib.gis.gdal.CoordTransform` object.
```
Comment (claudep) [inline docs/ref/contrib/gis/geos.txt]: A `versionchanged` directive would be nice in addition to this.

File: django/contrib/gis/geos/geometry.py [hunk 2]
```diff
@@ -494,14 +494,17 @@ def transform(self, ct, clone=False):
             else:
                 return
 
-        if (srid is None) or (srid < 0):
+        if isinstance(ct, gdal.CoordTransform):
+            # we don't care about SRID because CoordTransform presuppose source SRS
```
Comment (claudep) [inline django/contrib/gis/geos/geometry.py]: In new comments, we generally consider them sentences (uppercase first letter, dot at the end).

File: docs/ref/contrib/gis/geos.txt [hunk 3]
```diff
@@ -554,6 +554,11 @@ is returned instead.
     Requires GDAL. Raises :class:`~django.contrib.gis.geos.GEOSException` if
     GDAL is not available or if the SRID is ``None`` or less than 0.
 
+    .. versionchanged:: 1.9
```
Comment (timgraham) [inline docs/ref/contrib/gis/geos.txt]: The annotation should be "self-contained" as described in [Documenting new features](https://docs.djangoproject.com/en/dev/internals/contributing/writing-documentation/#documenting-new-features). Also, we are targeting 1.10 for new features and non-release blocking bug fixes (as far as I can tell this isn't one).

**Comment by timgraham** [comment]
merged in a7bb5af50be7b0efbb784011a41fd25fdfe4d42c, thanks!

**Pull request closed without merging**